langkit/extract.py

```diff
@@ -0,0 +1,16 @@
+import pandas as pd
+from typing import Any, Dict, Optional
+from whylogs.experimental.core.udf_schema import udf_schema
+
+
+def extract(
+    pandas: Optional[pd.DataFrame] = None,
+    row: Optional[Dict[str, Any]] = None,
```

Comment:
Can we make this a single input parameter for the data and we check for the shape like we do in whylogs?

Also consider optional param for DatasetSchema and use udf_schema() only if schema is None